src/leaflet.geometryutil.js

```diff
@@ -157,6 +157,36 @@ L.GeometryUtil = L.extend(L.GeometryUtil || {}, {
         return map.unproject(closest, maxzoom);
     },
 
+    /**
+    Returns the closest point of a {L.LatLng} on a {L.Circle}
+
+    @tutorial closest
+
+    @param {L.LatLng} latlng - The position to search
+    @param {L.Circle} circle - A Circle defined by a center and a radius
+    @returns {L.LatLng} Closest geographical point on the circle circumference
+    */
+    closestOnCircle: function (circle, latLng) {
+        const center = circle.getLatLng();
+        const circleRadius = circle.getRadius();
+        const radius = typeof circleRadius === 'number' ? circleRadius : circleRadius.radius;
+        const x = latLng.lng;
+        const y = latLng.lat;
+        const cx = center.lng;
+        const cy = center.lat;
+        const dx = x - cx;
+        const dy = y - cy;
+        const distanceSquared = dx * dx + dy * dy;
+
+        const distance = Math.sqrt(distanceSquared)
+
+        const tx = cx + (dx / distance) * radius;
+        const ty = cy + (dy / distance) * radius;
```

Comment:
Can you add a comment explaining what you are doing here?
(eg. cross multiplication between the distance of the point to the center and the radius)

Reply:
I added some comments